akka-http-core/src/main/scala/akka/http/scaladsl/Http.scala

```diff
@@ -258,7 +258,8 @@ class HttpExt(private val config: Config)(implicit val system: ActorSystem) exte
   def clientLayer(hostHeader: Host,
                   settings: ClientConnectionSettings,
                   log: LoggingAdapter = system.log): ClientLayer =
-    OutgoingConnectionBlueprint(hostHeader, settings, log)
+    RedirectSupportStage(settings.redirectSettings).atop(OutgoingConnectionBlueprint(hostHeader, settings, log))
```

Comment:
I thought about those other aspects as well. Maybe it makes sense to introduce something like `HighLevelHttpClient`, to wrap all those aspects and conditionally enable or disable them separately based on the settings? Of course, this potential `HighLevelHttpClient` is bigger than this PR, and there's already an issue for it: https://github.com/akka/akka/issues/16856. I just think this particular place, i.e. `atop(OutgoingConnectionBlueprint)`, is the most appropriate place to add such high-level client.

Every such feature incur some overhead, and client doesn't have to pay the price for something he/she doesn't need, like these redirects, for example. In this particular case, I think I can check the settings on both `cross-origin` and `same-origin` branches, and if both are disabled, don't add this `RedirectSupportStage` on top of the `OutgoingConnectionBlueprint` at all.